src/promptflow-core/promptflow/executor/flow_executor.py

```diff
@@ -874,6 +874,16 @@ def _exec_inner_with_trace(
             span.set_status(StatusCode.OK)
             return output, aggregation_inputs
 
+    @contextlib.contextmanager
+    def _record_keyboard_interrupt_to_span(self, span: Span):
+        try:
+            yield
+        except KeyboardInterrupt as ex:
+            if span.is_recording():
+                span.record_exception(ex)
+                span.set_status(StatusCode.ERROR, "KeyboardInterrupt received, execution cancelled.")
```

Comment:
KeyboardInterrupt is our internal implementation for all cases except when user actually hit ctrl-c locally.
The message should just be "Excecution cancelled."